Maker-Agents-Suit-Module/ESMAgent/src/ESM.join.event.agent.ts

```diff
@@ -10,24 +11,31 @@ import {
   FindingType,
   Log,
 } from 'forta-agent';
+import { decodeParam } from './utils';
 
 export const MAKER_ESM_JOIN_EVENT_SIGNATURE = 'Join(address,uint256)';
 export const MKR_DECIMALS = 18;
+export const MAKER_EVEREST_ID = '0xbabb5eed78212ab2db6705e6dfd53e7e5eaca437';
 
 const filterLog = (log: Log): boolean => {
-  const value = BigInt(log.data) / BigInt(10 ** MKR_DECIMALS);
-
-  return value > 2;
+  const amount = decodeParam('uint256', log.data);
+  return amount > 2;
 };
 
-const createFindingGenerator = (alertID: string): FindingGenerator => {
-  return () =>
+const createFindingGenerator = (_alertID: string): FindingGenerator => {
+  return (metadata: { [key: string]: any } | undefined): Finding =>
     Finding.fromObject({
-      name: 'Maker ESM Contract Join Detect Agent',
+      name: 'Maker ESM Join Event',
       description: 'Greater than 2 MKR is sent to ESM contract.',
-      alertId: alertID,
+      alertId: _alertID,
+      protocol: 'Maker',
       severity: FindingSeverity.Medium,
-      type: FindingType.Unknown,
+      type: FindingType.Suspicious,
+      everestId: MAKER_EVEREST_ID,
+      metadata: {
+        usr: decodeParam('address', metadata!.topics[1]).toLowerCase(),
+        amount: decodeParam('uint256', metadata!.data).toLowerCase(),
```

Comment:
You can remove the `.toLowerCase()` in `amount`.
```suggestion
        amount: decodeParam('uint256', metadata!.data),
```